helper/resource/testing_new.go

```diff
@@ -143,10 +144,13 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 	}
 }
 
-func getState(t testing.T, wd *tftest.WorkingDir) *terraform.State {
+func getState(t testing.T, wd *plugintest.WorkingDir) *terraform.State {
 	t.Helper()
 
-	jsonState := wd.RequireState(t)
+	jsonState, err := wd.State()
+	if err != nil {
+		t.Fatalf("Error reading state: %s", err)
```

Comment:
I'm ok with merging this as is and splitting this out as a separate task, but I think we should return this error instead of essentially panicking on it, for all the reasons we want to stop using Require: we sometimes want to add decorative text to that error, maybe about test step numbers, maybe about dangling resources, and `t.Fatal` robs us of that opportunity.

Reply:
Yes, will change these